### PR TITLE
CXX-847 Document the model/write class

### PR DESCRIPTION
**Pull request by acmorrow**

### CODE REVIEW

File: src/mongocxx/model/write.hpp [hunk 1]
```diff
@@ -32,32 +32,105 @@ MONGOCXX_INLINE_NAMESPACE_BEGIN
...
+    ///
     write(delete_many value);
+
+    ///
+    /// Constructs a write from a @replace_one.
+    ///
     write(replace_one value);
 
+    ///
+    /// Move constructs a write
```
Comment (Machyne) [inline src/mongocxx/model/write.hpp]: End with a `.`

File: src/mongocxx/model/write.hpp [hunk 2]
```diff
@@ -32,32 +32,105 @@ MONGOCXX_INLINE_NAMESPACE_BEGIN
...
+    ///
     write(write&& rhs) noexcept;
+
+    ///
+    /// Move assigns a write
```
Comment (Machyne) [inline src/mongocxx/model/write.hpp]: End with a `.`

File: src/mongocxx/model/write.hpp [hunk 3]
```diff
@@ -32,32 +32,105 @@ MONGOCXX_INLINE_NAMESPACE_BEGIN
...
+    ///
+    /// Returns the current type of this write. You must call this
+    /// method before calling any of the get methods below.
+    ///
     write_type type() const;
 
+    ///
+    /// Accesses the write as an @insert_one. It is illegal to call
+    /// this method if the type returned by the type method above does
+    /// not indicate that it is the proper type.
```
Comment (Machyne) [inline src/mongocxx/model/write.hpp]: The repeated:

> It is illegal to call this method if the type returned by the type method above does not indicate that it is the proper type.

Is a little hard to follow. Maybe:

> It is illegal to call this method if the return of @type (above) does not indicate that <insert_one e.g.> is the proper type of this write model.

File: src/mongocxx/model/write.hpp (hunk 1 above)
Comment (acmorrow) [inline src/mongocxx/model/write.hpp]: Done

File: src/mongocxx/model/write.hpp (hunk 2 above)
Comment (acmorrow) [inline src/mongocxx/model/write.hpp]: Done

File: src/mongocxx/model/write.hpp (hunk 3 above)
Comment (acmorrow) [inline src/mongocxx/model/write.hpp]: Done, more or less. See what you think of the new language.

**Comment by acmorrow** [comment]
Thanks for the review @Machyne. Updated, PTAL.

File: src/mongocxx/model/write.hpp (hunk 3 above)
Comment (Machyne) [inline src/mongocxx/model/write.hpp]: I think this way is more clear. Nice!

**Comment by Machyne** [comment]
LGTM! 🚢
